File: 2022/05/16/missingNumber.py
"""
https://leetcode.com/problems/missing-number/
Author: Andrew Jarombek
Date: 5/16/2022
"""

from typing import List


def missing_number(nums: List[int]) -> int:
    unique_nums = set(nums)

    for i in range(0, len(nums) + 1):
        if i not in unique_nums:
            return i

    return -1

File: 2023/02/04/go/missing_number.go
/**
 * https://leetcode.com/problems/missing-number/
 * Author: Andrew Jarombek
 * Date: 2/4/2023
 */

package _go

func missingNumber(nums []int) int {
	set := make(map[int]struct{})

	for _, num := range nums {
		set[num] = struct{}{}
	}

	for i := 0; i <= len(nums); i++ {
		if _, ok := set[i]; !ok {
			return i
		}
	}

	return -1
}
